lib/rubocop/cop/rspec/describe_class.rb

```diff
@@ -27,36 +40,37 @@ class DescribeClass < Base
         MSG = 'The first argument to describe should be '\
               'the class or module being tested.'
 
-        def_node_matcher :rails_metadata?, <<-PATTERN
-          (pair
-            (sym :type)
-            (sym { :channel :controller :helper :job :mailer :model :request
-                   :routing :view :feature :system :mailbox })
-          )
-        PATTERN
-
-        def_node_matcher :example_group_with_rails_metadata?, <<~PATTERN
-          (send #rspec? :describe ... (hash <#rails_metadata? ...>))
+        def_node_matcher :example_group_with_ignored_metadata?, <<~PATTERN
+          (send #rspec? :describe ... (hash <#ignored_metadata? ...>))
         PATTERN
 
         def_node_matcher :not_a_const_described, <<~PATTERN
           (send #rspec? :describe $[!const !#string_constant?] ...)
         PATTERN
 
-        def on_top_level_group(top_level_node)
-          return if example_group_with_rails_metadata?(top_level_node.send_node)
+        def on_top_level_group(node)
+          return if example_group_with_ignored_metadata?(node.send_node)
 
-          not_a_const_described(top_level_node.send_node) do |described|
+          not_a_const_described(node.send_node) do |described|
             add_offense(described)
           end
         end
 
         private
 
+        def ignored_metadata?(node)
+          ignored_metadata.find { |key, _v| key.to_sym == node.key.value }
+            &.last&.map(&:to_sym)&.include? node.value.value
```

Comment:
Is it the same as?
```ruby
ignored_metadata[node.key.value.to_s].include?(node.value.value.to_s)
```
So you try to avoid string allocation for each metadata key and value?

By the way, what if metadata value is not a string or symbol literal?

```ruby
describe 'A', type: method_call do
```

or
```ruby
describe 'A', type: 'fea' + 'ture' do
```

I'd suggest narrowing this down to string and symbol literals.

Reply:
The only difference is that your snippet would ignore both symbols and strings, while currently the method only ignores symbols.

You're right, the code breaks in both of those cases. I added examples, validated for pairs of symbols and changed the code to your version. You want it to also work for strings? If so, can you think of a simpler way to do it than replicating the condition?